security/pkg/pki/ra/common.go

```diff
@@ -86,9 +91,78 @@ func ValidateCSR(csrPEM []byte, subjectIDs []string) bool {
 			return false
 		}
 	}
+
+	// Verify Istio CSR was created by Istio-Proxy by generating template using same the
+	// same process as performed on the client and comparing the generated CSR with the
+	// received CSR.
+
+	// Convert csrIDs to Host comma separated string
```

Comment:
Technically - agent never sends a comma separated list. I have a proposal to add a DNS SAN for interop - but it doesn't seem very popular and it'll likely be better done with an external CA.